app/jobs/kithe/create_derivatives_job.rb

```diff
@@ -1,7 +1,13 @@
 module Kithe
   class CreateDerivativesJob < Job
     def perform(asset, mark_created: nil)
-      asset.create_derivatives(mark_created: mark_created)
+        asset.create_derivatives(mark_created: mark_created)
+    end
+    # This error typically occurs when several large assets, whose derivatives
+    # take a long time to generate, are deleted immediately after ingest.
+    rescue_from(ActiveJob::DeserializationError) do |exception|
+      Rails.logger.error("Unable to derivatives for this asset, as it was unavailable.")
+      Rails.logger.error("Details: #{exception.inspect}")
```

Comment:
Two separate `Rails.logger.error` lines risks that some other log entry could end up being logged in between them. Better to do one single `Rails.logger.error`. If you want multiple lines, you can still put a `"\n"` in there. But i'm not sure it needs to be multiple lines, what about just `Unable to delete deravatives for this asset, as it was unavailable: #{exception}"`? (Not sure what difference adding explicit `.inspect` makes if any). (Note also missing verb). 

Thanks for pasting the stack trace from the uncaught exception. But what I really want to see is what the *log* looks like from *this* code, what it actually looks like when you log it.